Source/UI/NewUI/PaymentForm/View/SecurityCodeView.swift

```diff
@@ -39,20 +45,38 @@ public final class SecurityCodeView: UIView {
     codeInputView.update(style: self.style)
   }
 
-  private func updateErrorViewStyle(isHidden: Bool, textfieldText: String?) {
+  // TODO: integrate with payment vc when card view is finished
+  func updateCardSchema(cardSchema: Card.Scheme) {
+    supportedSchema = cardSchema
+  }
+
+  private func updateErrorViewStyle(isHidden: Bool, textfieldText: String?){
     style?.error?.isHidden = isHidden
     style?.textfield.text = textfieldText ?? ""
     codeInputView.update(style: style)
   }
+
+  func validateSecurityCode(with text: String?) {
+    guard let text = text?.filter { !$0.isWhitespace }, !text.isEmpty else {
+      updateErrorViewStyle(isHidden: false, textfieldText: text)
+      return
+    }
+    switch cardValidator.validate(cvv: text, cardScheme: supportedSchema)  {
+      case .success:
+        updateErrorViewStyle(isHidden: true, textfieldText: text)
+        delegate?.update(securityCode: text)
+      case .failure(let error):
+        print(error)
+        updateErrorViewStyle(isHidden: false, textfieldText: text)
```

Comment:
If an error is received, should we still inform delegate?

Reply:
No, no need for now. only update on success for more than or equal 3